src/gui/filedetails/sharemodel.cpp

```diff
@@ -766,6 +767,33 @@ void ShareModel::slotShareExpireDateSet(const QString &shareId)
     Q_EMIT dataChanged(shareModelIndex, shareModelIndex, { ExpireDateEnabledRole, ExpireDateRole });
 }
 
+void ShareModel::slotDeleteE2EeShare(const SharePtr &share) const
+{
+    const auto account = accountState()->account();
+    QString folderAlias;
+    for (const auto &f : FolderMan::instance()->map()) {
+        if (f->accountState()->account() != account) {
+            continue;
+        }
+        const auto folderPath = f->remotePath();
+        if (share->path().startsWith(folderPath) && (share->path() == folderPath || folderPath.endsWith('/') || share->path()[folderPath.size()] == '/')) {
+            folderAlias = f->alias();
```

Comment:
most probably you want to add a `break` after this line to avoid running the loop after you found the `folderAlias`